fw/vhost.c

```diff
@@ -1874,26 +1874,20 @@ tfw_vhost_add(TfwVhost *vhost)
  * Called on (re-)configuration time in process context.
  */
 void
-tfw_vhost_add_sni_map(const BasicStr *cn, const char *hname, int hlen)
+tfw_vhost_add_sni_map(const BasicStr *cn, TfwVhost *vhost)
 {
 	unsigned long key = basic_hash_str(cn);
 	TfwSVHMap *svhm;
 	int n = sizeof(*svhm) + cn->len;
-	const BasicStr ns = {.data = (char *)hname, .len = hlen};
 
 	if (!(svhm = kmalloc(n, GFP_KERNEL))) {
-		T_WARN("Cannot allocate mapping for SAN/CN %.*s -> %s\n",
-		       (int)cn->len, cn->data, hname);
+		T_WARN("Cannot allocate mapping for SAN/CN %.*s -> %.*s\n",
+		       (int)cn->len, cn->data,
+		       (int)vhost->name.len, vhost->name.data);
 		return;
 	}
 
-	svhm->vhost = __tfw_vhost_lookup(tfw_vhosts_reconfig, &ns,
```

Comment:
@krizhanovsky @const-t , can I use vhost pointer instead of a name lookup? Name lookup fails to find vhost "default".

Reply:
Why do we call `tfw_vhost_add_sni_map` for `default` vhost? As I understand `default` vhost must be used ONLY if user specified directive `allow_any_sni` and SNI doesn't match to any SAN or vhost name. If we call `tfw_vhost_add_sni_map` for `default` we assume that we can receive SNI = `default`(in this case `tfw_tls_sni` will block such handshake) or certificate's SAN can contain `default` name that strange.